### PR TITLE
app-server: add v2 filesystem APIs

### PR DESCRIPTION
**Pull request by euroelessar**
Add a protocol-level filesystem surface to the v2 app-server so Codex clients can read and write files, inspect directories, and subscribe to path changes without relying on host-specific helpers.

High-level changes:
- define the new v2 fs/readFile, fs/writeFile, fs/createDirectory, fs/getMetadata, fs/readDirectory, fs/remove, fs/copy RPCs
- implement the app-server handlers, including absolute-path validation, base64 file payloads, recursive copy/remove semantics
- document the API, regenerate protocol schemas/types, and add end-to-end tests for filesystem operations, copy edge cases

Testing plan:
- validate protocol serialization and generated schema output for the new fs request, response, and notification types
- run app-server integration coverage for file and directory CRUD paths, metadata/readDirectory responses, copy failure modes, and absolute-path validation

### CODE REVIEW

File: codex-rs/app-server/src/fs_api.rs [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+#[derive(Clone, Default)]
+pub(crate) struct FsApi;
+
+impl FsApi {
+    pub(crate) async fn read_file(
+        &self,
+        params: FsReadFileParams,
+    ) -> Result<FsReadFileResponse, JSONRPCErrorError> {
+        require_absolute_path(&params.path, "fs/readFile", "path")?;
+        let bytes = tokio::fs::read(params.path).await.map_err(map_io_error)?;
```
[the remark this answers is not in the document]
Comment (euroelessar) [reply codex-rs/app-server/src/fs_api.rs:37]: that's fine for now, if needed we can introduce streaming-based apis for read/write later on (and limit message sizes for non-streaming ones)

File: codex-rs/app-server/src/fs_api.rs [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+                "fs/copy only supports regular files, directories, and symlinks",
+            ))
+        })
+        .await
+        .map_err(map_join_error)??;
+        Ok(FsCopyResponse {})
+    }
+}
+
+fn copy_dir_recursive(source: &Path, target: &Path) -> io::Result<()> {
```
Comment (owenlin0) [inline codex-rs/app-server/src/fs_api.rs:195]: consider using the `walkdir` crate to navigate a directory tree, we use it elsewhere in the repo too

**Review by owenlin0 — APPROVED** [review]
